juju/juju
# Juju 1678 add jammy to juju qa test 2.9

This is a rebase of https://github.com/juju/juju/pull/14515 onto the 2.9 branch. 
We updated the juju-qa-test charm which effects both 2.9 CI as well as 3.0 CI, so these updates should be targeted to the 2.9 branch.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*


## QA steps

The biggest thing to verify is that the test suites match the revisions that are listed in README.md. 
```sh
$ juju info juju-qa-test
...
channels: |
  latest/stable:     1.1-stable  2022-08-24  (19)  2MB
  latest/candidate:  1.4-cand    2022-08-24  (20)  2MB
  latest/beta:       ↑
  latest/edge:       2.0-edge    2022-08-24  (21)  2MB
  2.0/stable:        2.0-stable  2022-08-26  (22)  2MB
  2.0/candidate:     ↑
  2.0/beta:          ↑
  2.0/edge:          2.0-edge  2022-08-29  (23)  6MB
```
Should match the version strings and revision ids in README.md. And with a deeper look, should also match the revisions in various tests associated with this.

But also the guidelines from the Readme about trying to build and deploy a charm:
```sh
cd testcharms/charm-hub/charms/juju-qa-test
make stable
juju deploy ./juju-qa-test_ubuntu*.charm
```

## Documentation changes

None

## Bug reference

None
